log/codecomet_writer.go

```diff
@@ -0,0 +1,471 @@
+package log
```

Comment:
i tried reducing this down some, but unfortunately the way it behaves (and go's attitude to inheritance) makes it fairly impossible. I don't love that it's mostly a clone of `ConsoleWriter`, but overall this seems somewhat smarter than forking `zerolog`.